pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -278,6 +282,60 @@ type TiKVSpec struct {
 	Config *TiKVConfig `json:"config,omitempty"`
 }
 
+// TiFlashSpec contains details of TiFlash members
+// +k8s:openapi-gen=true
+type TiFlashSpec struct {
+	ComponentSpec               `json:",inline"`
+	corev1.ResourceRequirements `json:",inline"`
+
+	// Specify a Service Account for TiFlash
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+
+	// The desired ready replicas
+	// +kubebuilder:validation:Minimum=1
+	Replicas int32 `json:"replicas"`
+
+	// Base image of the component, image tag is now allowed during validation
+	// +kubebuilder:default=pingcap/tiflash
+	// +optional
+	BaseImage string `json:"baseImage"`
+
+	// Whether create the TiFlash container in privileged mode, it is highly discouraged to enable this in
+	// critical environment.
+	// Optional: defaults to false
+	// +optional
+	Privileged *bool `json:"privileged,omitempty"`
+
+	// MaxFailoverCount limit the max replicas could be added in failover, 0 means no failover
+	// Optional: Defaults to 3
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MaxFailoverCount *int32 `json:"maxFailoverCount,omitempty"`
+
+	// The DiskRequests of the persistent volume for TiFlash data storage.
+	// TiFlash supports multiple disks
+	DiskRequests []DiskRequests `json:"diskRequests"`
+
+	// Config is the Configuration of TiFlash
+	// +optional
+	Config *TiFlashConfig `json:"config,omitempty"`
+
+	// Dedicated TiDB cluster spec for TiFlash
+	TiDB *TiDBSpec `json:"tidb,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// TiDBSpec contains details of TiDB members
+type DiskRequests struct {
+	// The storageClassName of the persistent volume for data storage.
+	// Defaults to Kubernetes default storage class.
+	// +optional
+	StorageClassName *string `json:"storageClassName,omitempty"`
+
+	// StorageSize is the request storage size for TiFlash
+	StorageSize string `json:"storageSize"`
```

Comment:
We should keep the expression as same as `pd` and `tikv`, so i tend to use `storage` here.

Reply:
The `storage` for pd and tikv is configured in `corev1.ResourceRequirements` where the other fields are like `cpu`, `memory`, etc. However, `storage` here will not be clear enough.

Reply:
Event if assuming the `storage` is not clear enough, I think there is no need to make difference between `pd`/`tikv`/`tiflash` and kubernetes resources expression. `StorageSize` seems like the property to dedicate the byte size of the storage requests which is common used in `TiDB`.

Reply:
how about using a more generic and Kubernetes-style to configure volumes starting from tiflash? we can use volumeClaimTemplate to configure volumes for container and volumeMount to configure how to mount them. then we can use all functions they provide, e.g.

- `volumeMode`, `selector` from `volumeClaimTemplate`.
- `subPath` from `volumeMount` (https://github.com/pingcap/tidb-operator/issues/1525#issuecomment-602016020)

Reply:
Good idea!

Reply:
Updated.